crates/flox/doc/flox-publish.md

```diff
@@ -10,12 +10,30 @@ flox-publish - build and publish project to flox channel
 
 # SYNOPSIS
 
-flox [ `<general-options>` ] publish [ `<options>` ]
+flox [ `<general-options>` ] publish [ `<options>` ] [`<package>`]
 
 # DESCRIPTION
 
-Perform a build, (optionally) copy binaries to a cache,
-and add package metadata to a flox channel.
+Adds metadata to a catalog for the package identified by `<package>`
+or dynamically selected from a choice of packages in the current directory.
+
+Prior to to submitting the metadata,
+it will verify that the package can be built from its _upstream_ location
+and optionally sign and cache the resulting binary.
+
+The package must be defined in a remote git repository and be referred to
+either directly by a `git+ssh://<url>[#<package>]` url or another url that
+can be resolved to an upstream git resource.
```

Comment:
needs some note about the heuristic that the remote origin is used

Reply:
```suggestion
can be resolved to an upstream git resource.
Packages in local repositories are resolved to the current branch's upstream branch.
```